src/selector_gui.cpp

```diff
@@ -0,0 +1,386 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file selector_gui.cpp A composable selector widget. */
+
+#include "3rdparty/fmt/core.h"
+#include "stdafx.h"
+#include "company_base.h"
+#include "company_gui.h"
+#include "strings_func.h"
+#include "table/sprites.h"
+#include "selector_gui.h"
+#include "widgets/selector_widget.h"
+#include "safeguards.h"
+
+
+/**
+ * Create a new NWidgetBase with all the widgets needed by the SelectorWidget widget
+ * This function is made to be passed in NWidgetFunction, when creating UI for a parent window
+ * @see NWidgetFunction
+ */
+std::unique_ptr<NWidgetBase> SelectorWidget::MakeSelectorWidgetUI()
+{
+	static constexpr NWidgetPart widget_ui[] = {
+		NWidget(WWT_PANEL, COLOUR_BROWN),
+			NWidget(WWT_EDITBOX, COLOUR_BROWN, WID_SELECTOR_EDITBOX), SetFill(1, 0), SetResize(1, 0), SetPadding(2),
+			SetDataTip(STR_LIST_FILTER_TOOLTIP, STR_LIST_FILTER_TOOLTIP),
```

Comment:
Indenting is still wrong for this whole widget. Continued lines should be two tabs deeper than the line they continue from.

```suggestion
			NWidget(WWT_EDITBOX, COLOUR_BROWN, WID_SELECTOR_EDITBOX), SetFill(1, 0), SetResize(1, 0), SetPadding(2),
					SetDataTip(STR_LIST_FILTER_TOOLTIP, STR_LIST_FILTER_TOOLTIP),
```

This one could also be one line.

```suggestion
			NWidget(WWT_EDITBOX, COLOUR_BROWN, WID_SELECTOR_EDITBOX), SetFill(1, 0), SetResize(1, 0), SetPadding(2), SetDataTip(STR_LIST_FILTER_TOOLTIP, STR_LIST_FILTER_TOOLTIP),
```